### PR TITLE
added trace() and alphas_trace() methods to PDF class

### PR DESCRIPTION
**Pull request by marcorossi5**
Now when a PDF object is instantiated it's sufficient to call the trace() or alphas_trace() methods in order to build the graphs automatically.
I changed the benchmark accordingly.
Note that when running benchmarks, the printed "Building graph time" to stdout is not just a single graph tracing as before, but all the possible PDF python UI are called. Then this time could be higher than before.

### CODE REVIEW

File: src/pdfflow/pflow.py [hunk 1]
```diff
@@ -488,3 +488,82 @@ def py_alphasQ(self, arr_q):
 
         # Perform the actual computation
         return self.alphasQ(a_q)
+
+    def trace(self):
+        """
+        Builds all the needed graph in advance
+        of interpolations
+        """
+        print('Building tf.Graph ...')
```
Comment (scarlehoff) [inline src/pdfflow/pflow.py:497]: The print statement should be `logger.info`

File: src/pdfflow/pflow.py [hunk 2]
```diff
@@ -488,3 +488,82 @@ def py_alphasQ(self, arr_q):
...
+        x = []
+        q2 = []
+
+        #x points are equal for all the subgrids
+        xmin = float(tf.math.exp(self.subgrids[0].log_xmin))
+
+        q2min = float(tf.math.exp(self.subgrids[0].log_q2min))
```
Comment (scarlehoff) [inline src/pdfflow/pflow.py:504]: Instead of casting it to float, I think it would be better to do `.numpy()` when possible.

**Review by scarlehoff — COMMENTED** [review]
This is actually quite nice (sorry I only reviewed it now).
The only thing missing is some documentation about the what and the why.

Maybe a new section in the [How to Use](https://pdfflow.readthedocs.io/en/latest/how_to.html) titled for instance: "Building the graph ahead of time" (for that you need to modify this file: https://github.com/N3PDF/pdfflow/blob/master/doc/source/how_to.rst )

**Comment by marcorossi5** [comment]
How can I make a newline inside a code block in a rst file?

**Comment by scarlehoff** [comment]
I think inside a code block the breaklines are taken at face value. So it should work if you write the code normally

**Review by scarlehoff — APPROVED** [review]
I think this can be merged, so that some of the other PR can also be merged and then we can polish the corners (mainly docs) in different PRs (this one is growing out of scope, specially the benchmarks).

**Pull request merged**